README.md

```diff
@@ -15,6 +47,507 @@ go get -u github.com/src-d/kallax/...
+
+* `BeforeInsert`: will be called before inserting the model.
+* `BeforeUpdate`: will be called before updating the model.
+* `BeforeSave`: will be called before updating or inserting the model. It's always called before `BeforeInsert` and `BeforeUpdate`.
+* `BeforeDelete`: will be called before deleting the model.
+* `AfterInsert`: will be called after inserting the model. The presence of this event will cause the insertion of the model to run in a transaction. If the event returns an error, it will be rolled back.
+* `AfterUpdate`: will be called after updating the model. The presence of this event will cause the update of the model to run in a transaction. If the event returns an error, it will be rolled back.
+* `AfterSave`: will be called after updating or inserting the model. It's always called after `AfterInsert` and `AfterUpdate`. The presence of this event will cause the operation with the model to run in a transaction. If the event returns an error, it will be rolled back.
+* `AfterDelete`: will be called after deleting the model. The presence of this event will cause the deletion to run in a transaction. If the event returns an error, it will be rolled back.
+
+To implement these events, just implement the following interfaces. You can implement as many as you want:
+
+* [BeforeInserter](https://godoc.org/github.com/src-d/go-kallax#BeforeInserter)
+* [BeforeUpdater](https://godoc.org/github.com/src-d/go-kallax#BeforeUpdater)
+* [BeforeSaver](https://godoc.org/github.com/src-d/go-kallax#BeforeSaver)
+* [BeforeDeleter](https://godoc.org/github.com/src-d/go-kallax#BeforeDeleter)
+* [AfterInserter](https://godoc.org/github.com/src-d/go-kallax#AfterInserter)
+* [AfterUpdater](https://godoc.org/github.com/src-d/go-kallax#AfterUpdater)
+* [AfterSaver](https://godoc.org/github.com/src-d/go-kallax#AfterSaver)
+* [AfterDeleter](https://godoc.org/github.com/src-d/go-kallax#AfterDeleter)
+
+Example:
+
+```go
+func (u *User) BeforeSave() error {
+        if u.Password == "" {
+                return errors.New("cannot save user without password")
+        }
+
+        if !isCrypted(u.Password) {
+                u.Password = crypt(u.Password)
+        }
+        return nil
+}
+```
+
+## Kallax generated code
+
+Kallax generates a bunch of code for every single model you have and saves it to a file named `kallax.go` in the same package.
+
+For every model you have, kallax will generate the following for you:
+
+* Internal methods for your model to make it work with kallax and satisfy the [Record](https://godoc.org/github.com/src-d/go-kallax#Record) interface.
+* A store named `{TypeName}Store`: the store is the way to access the data. A store of a given type is the way to access and manipulate data of that type.
+* A query named `{TypeName}Query`: the query is the way you will be able to build programmatically the queries to perform on the store. A store only will accept queries of its own type.
```

Comment:
I'd explain (here, or in the following query section), that adding filters to the query modifies the query itself (it was not obvious when we landed at squirrel for the first time and we needed to confirm it, so this clarification in advance would be useful for our new users)